### PR TITLE
fix(codebases): Headers must be on own list view row

### PR DESCRIPTION
**Pull request by dlabrecq**
Refactored list view to place column headers on their own row. 

An issue was discovered after adding an info icon to the list view header on the codebases page. When the user hovers over the icon, a tooltip is displayed, but the first row is also unexpectedly highlighted. Moving headers to their own row ensures mouseover/mouseenter events are not fired for the first list view row.

Note: This PR will depend on ngx-widgets:
https://github.com/fabric8-ui/ngx-widgets/pull/76

screen shot 2017-04-28 at 2 25 54 am

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 17ae22c1e2a7cf64196b36d4fe62b6ba1d332bb7
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by joshuawilson** [comment]
[test]

**Comment by catrobson** [comment]
It looks like we ended up with some typos/grammatical problems in the pop-up.  Remove "to" in "for to your code and runtime needs"

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: b02f6af174c36f7317d179fc45ae593dba080683
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c35969e70c65cb08f7eb9c9de2b43a4abf3202b5
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by fabric8cd** [comment]
@dlabrecq snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1200-fabric8-ui.int.rdu2c.fabric8.io

**Comment by catrobson** [comment]
Not sure if it is just me, but getting some odd issues now on Codebases screen in that snapshot:
image

**Comment by dlabrecq** [comment]
Unfortunately, this PR depends on changes made to the ngx-widgets repo. Until that is merged, the snapshot is invalid because it won't have the code it needs to render correctly.

**Comment by fabric8cd** [comment]
@dlabrecq snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1200-fabric8-ui.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@dlabrecq snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1200-fabric8-ui.int.rdu2c.fabric8.io

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 1907c6ee2afe99b34304b26ac585bad78059be31
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 009c9b63bd7f97d6abd3c2a23f85be5e8bea91af
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by fabric8cd** [comment]
@dlabrecq snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1200-fabric8-ui.int.rdu2c.fabric8.io

File: src/app/create/codebases/codebases-item-actions/codebases-item-actions.component.scss [hunk 1]
```diff
@@ -1,2 +1,9 @@
 @import '../../../../assets/stylesheets/base';
-codebases-item-actions { display: flex; }
+
+codebases-item-actions {
+  display: flex;
```
Comment (joshuawilson) [inline src/app/create/codebases/codebases-item-actions/codebases-item-actions.component.scss]: Based on the standard that @mindreeper2420 has started us using this should be on one line if there is only one attribute.

File: src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss [hunk 2]
```diff
@@ -1,22 +1,22 @@
 @import '../../../../assets/stylesheets/base';
-codebases-item-workspaces { display: flex; }
+
+codebases-item-workspaces {
```
Comment (joshuawilson) [inline src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss]: Please keep to one line for one attribute. It is better for debugging.

File: src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
+  display: flex;
+}
+
 .busy-indicator {
-  position: relative;
   display: flex;
   height: auto;
-  width: 25px;
   margin-left: 10px;
+  position: relative;
```
Comment (joshuawilson) [inline src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss]: The order of the attributes effects the rendering according to the research @mindreeper2420 has done. Please reset this.

File: src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
   vertical-align: center;
+  width: 25px;
   > div {
-    margin: auto;
-    padding: 0;
+    margin:auto;
```
Comment (joshuawilson) [inline src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss]: Why did you take the space out of this?

File: src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss (hunk 3 above)
Comment (dlabrecq) [inline src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss]: Reset what?

Comment (dlabrecq) [inline src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss]: What is the proposed declaration order? I've been following the Patternfly style guide below.

http://codeguide.patternfly.org/#css-declaration-order

I spoke with @mindreeper2420 and he wasn't aware that this guide existed prior to making changes -- we should discuss.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: da1ca6bc7174eb45d5d2380fae47c8ff3aba9679
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by fabric8cd** [comment]
@dlabrecq snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1200-fabric8-ui.int.rdu2c.fabric8.io

File: src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss (hunk 3 above)
Comment (mindreeper2420) [inline src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss]: @dlabrecq @joshuawilson so the PatternFly code guide is the same code guide that I attributed the changes to, but PatternFly forked it and has not updated to the latest version.

Comment (dlabrecq) [inline src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss]: It's still not clear what style guide you're referring to -- Bootstrap 4? Can you please provide a pointer?

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 74c1bc52b2c9fa4f12430c53d5959b45650a34f3
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

File: src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss (hunk 3 above)
Comment (dlabrecq) [inline src/app/create/codebases/codebases-item-workspaces/codebases-item-workspaces.component.scss]: Done. Using the declaration order from http://codeguide.co/#css-declaration-order

**Comment by joshuawilson** [comment]
[test]

**Comment by dlabrecq** [comment]
@joshuawilson Can we restart the continuous-integration build? I don't understand why the script failed with "Pipe not connected"?

**Comment by joshuawilson** [comment]
[test]

**Comment by fabric8cd** [comment]
@dlabrecq snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1200-fabric8-ui.int.rdu2c.fabric8.io

**Pull request merged**